libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/store/EventStoreManagerTest.java

```diff
@@ -193,22 +193,28 @@ public void testDeleteMultipleEvents() throws Exception {
         final InstrumentationEvent event2 = createTestEvent();
         Assert.assertNotNull("Generated event stored should not be null", event2);
         final String eventId2 = event2.getEventId();
+        final InstrumentationEvent event3 = createTestEvent();
+        Assert.assertNotNull("Generated event stored should not be null", event3);
+        final String eventId3 = event3.getEventId();
         final List<InstrumentationEvent> genEvents = new ArrayList<InstrumentationEvent>();
         genEvents.add(event1);
         genEvents.add(event2);
+        genEvents.add(event3);
         storeManager.storeEvents(genEvents);
         final List<InstrumentationEvent> eventsBeforeDel = storeManager.fetchAllEvents();
         Assert.assertNotNull("List of events stored should not be null", eventsBeforeDel);
-        Assert.assertEquals("Number of events stored should be 2", 2, eventsBeforeDel.size());
-        Assert.assertTrue("Stored event should be the same as generated event", event1.equals(eventsBeforeDel.get(0)));
-        Assert.assertTrue("Stored event should be the same as generated event", event2.equals(eventsBeforeDel.get(1)));
+        Assert.assertEquals("Number of events stored should be 3", 3, eventsBeforeDel.size());
```

Comment:
Not assuming that fetchAllEvents() which calls book.getAllKeys() which relies on bookFolder.list(new FilenameFilter() {...}) returns events in the order they were inserted.

Reply:
Interesting! So, the previous implementation of flat files guaranteed order because each event was in its own file, if I'm not mistaken, but `PaperDB` doesn't, which makes sense.